diag_manager/fms_diag_axis_object.F90

```diff
@@ -0,0 +1,287 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @defgroup fms_diag_axis_object_mod fms_diag_axis_object_mod
+!> @ingroup diag_manager
+!! @brief fms_diag_axis_object_mod stores the diag_axis_object
+
+!> @file
+!> @brief File for @ref diag_axis_object_mod
+
+!> @addtogroup fms_diag_axis_object_mod
+!> @{
+module fms_diag_axis_object_mod
+  use mpp_domains_mod, only:  domain1d, domain2d, domainUG, mpp_get_compute_domain, CENTER, &
+                            & mpp_get_compute_domain
+  use platform_mod,    only:  r8_kind, r4_kind
+  use diag_data_mod,   only:  diag_atttype
+  use mpp_mod,         only:  FATAL, mpp_error
+  implicit none
+
+  PRIVATE
+
+  public :: diag_axis_t, diag_axis_init, set_subaxis
+  !> @}
+
+  !> @brief Type to hold the domain info for an axis
+  !! This type was created to avoid having to send in "Domain", "Domain2", "DomainUG" as arguments into subroutines
+  !! and instead only 1 class(fms_domain_t) argument can be send
+  !> @ingroup diag_axis_object_mod
+  type fms_domain_t
```

Comment:
this is a diag manager type, so it should be named `fmsDiagDomain_t` or `diagDomain_t`.  Using this name makes it seem like it comes from `fms_mod`.    The same is true for the types that extend this one.